modules/workflow-service-api/src/main/java/org/opencastproject/workflow/api/WorkflowServiceDatabaseImpl.java

```diff
@@ -162,6 +162,67 @@ public List<WorkflowInstance> getWorkflowInstances(int limit, int offset) throws
     }
   }
 
+  /**
+   * {@inheritDoc}
+   *
+   * @see WorkflowServiceDatabase#getLatestWorkflowInstances(int limit, int offset)
+   */
+  public List<WorkflowInstance> getLatestWorkflowInstances(int limit, int offset) throws WorkflowDatabaseException {
+
+    EntityManager em = null;
+    try {
+      em = emf.createEntityManager();
+
+      var query = em.createNamedQuery("Workflow.findLatest", WorkflowInstance.class);
+
+      String orgId = securityService.getOrganization().getId();
+      query.setParameter("organizationId", orgId);
+      query.setParameter("stateInstantiated", WorkflowInstance.WorkflowState.INSTANTIATED);
+      query.setParameter("stateRunning", WorkflowInstance.WorkflowState.RUNNING);
+      query.setParameter("statePaused", WorkflowInstance.WorkflowState.PAUSED);
+      query.setParameter("stateFailing", WorkflowInstance.WorkflowState.FAILING);
+      query.setMaxResults(limit);
+      query.setFirstResult(offset);
+      logger.debug("Requesting latest workflows using query: {}", query);
+      return query.getResultList();
+    } catch (Exception e) {
+      throw new WorkflowDatabaseException(e);
+    } finally {
+      if (em != null)
+        em.close();
+    }
+  }
+
+    /**
+   * {@inheritDoc}
+   *
+   * @see WorkflowServiceDatabase#getLatestWorkflowCount()
+   */
+  public int getLatestWorkflowCount() throws WorkflowDatabaseException {
+
+    EntityManager em = null;
+    try {
+      em = emf.createEntityManager();
+
+      Query query = em.createNamedQuery("Workflow.countLatest");
```

Comment:
So you don't need type casting later:

```suggestion
      var query = em.createNamedQuery("Workflow.countLatest", Long.class);
```